### PR TITLE
Adding tests for legacy checkpoints - 1.8.x

### PR DESCRIPTION
**Pull request by Borda**
## What does this PR do?

Have been missing for the whole 1.8 and 1.9 series
~the 1.9.5 is added in #15946~

<details>
  <summary><b>Before submitting</b></summary>

- Was this **discussed/agreed** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/Lightning-AI/lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- Did you make sure to **update the documentation** with your changes? (if necessary)
- Did you write any **new necessary tests**? (not for typos and docs)
- [x] Did you verify new and **existing tests pass** locally with your changes?
- Did you list all the **breaking changes** introduced by this pull request?
- Did you **update the CHANGELOG**? (not for typos, docs, test updates, or minor internal changes/refactors)

</details>

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing, make sure you have read the [review guidelines](https://github.com/Lightning-AI/lightning/wiki/Review-guidelines). In short, see the following bullet-list:

<details>
  <summary>Reviewer checklist</summary>

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [x] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

</details>

cc @borda @carmocca @awaelchli

### CODE REVIEW

**Comment by Borda** [comment]
cross-link to the same PR for LTS: #17373

**Review by ethanwharris — APPROVED** [review]

**Review by lantiga — APPROVED** [review]
👍

File: tests/tests_pytorch/helpers/simple_models.py [hunk 1]
```diff
@@ -11,13 +11,18 @@
...
+import operator
+
 import torch
 import torch.nn.functional as F
+from lightning_utilities.core.imports import compare_version
 from torch import nn
 from torchmetrics import Accuracy, MeanSquaredError
 
 from lightning.pytorch import LightningModule
-from lightning.pytorch.utilities.imports import _TORCHMETRICS_GREATER_EQUAL_0_11 as _TM_GE_0_11
```
Comment (carmocca) [inline tests/tests_pytorch/helpers/simple_models.py:20]: Was something not working with this import?

Comment (Borda) [reply tests/tests_pytorch/helpers/simple_models.py:20]: While building some past ckpts, the package does not have this variable
